orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/CIJobRequest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.igor.tasks;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+import java.util.Optional;
+
+@Getter
+public class CIJobRequest {
+  private final String master;
+  private final String job;
+  private final String propertyFile;
+  private final Integer buildNumber;
+  private final int consecutiveErrors;
+
+  public CIJobRequest(
+    @JsonProperty("master") String master,
```

Comment:
Is there a reason we have to annotate every param here? Wouldn't `@Data` and `@AllArgsConstructor` work?

Reply:
I could not get any combination of Lombok annotations to get me an immutable class that would properly deserialize with Jackson.  The release notes for 1.16.22 (the next one after ours) mention better compatibility with Jackson, but I didn't want to expand the scope here to upgrading Lombok.

I guess I could have made it non-immutable so it could use the no-args constructor and setters (which is what I think mostly happens in our code)...but given that it never needs to be mutated that didn't feel like a great solution.

I'll at least add a comment as to why I did this.

Reply:
:+1: to comment.